src/charm.py

```diff
@@ -176,6 +186,29 @@ def _on_cluster_relation_changed(self, event: EventBase) -> None:
         if self.state.cluster.switching_encryption and len(self.state.servers) == 1:
             event.defer()
 
+    def _on_secret_changed(self, event: SecretChangedEvent):
+        """Reconfigure services on a secret changed event."""
+        if not event.secret.label:
+            return
+
+        if not self.state.cluster.relation:
+            return
+
+        if event.secret.label == self.state.cluster.data_interface._generate_secret_label(
+            PEER,
+            self.state.cluster.relation.id,
+            None,  # type:ignore noqa  -- Changes with the soon upcoming new version of DP-libs STILL within this POC
+        ):
```

Comment:
NOTE: The `# type: ignore` should not be needed anymore once https://github.com/canonical/data-platform-libs/pull/137 is merged (may take a while :sweat_smile: )

Calling ain INTERNAL function is CLEARLY wrong, should be replaced with a public call once https://github.com/canonical/data-platform-libs/issues/124 is finally addressed.

(It's a small thing I just never gt to it...)